File: Backend/Travle.Model/Responses/FavoriteToggleResponse.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// The result of toggling a favorite: the new state after the toggle. <see cref="IsFavorite"/> is true
    /// when the target is now favorited, false when it was just removed — the client flips the heart from
    /// this rather than assuming the outcome.
    /// </summary>
    public class FavoriteToggleResponse
    {
        /// <summary>"Destination" or "Tour" — echoes which target was toggled.</summary>
        public string TargetType { get; set; } = string.Empty;
        public int TargetId { get; set; }

        /// <summary>True if the target is now in the user's favorites, false if it was just removed.</summary>
        public bool IsFavorite { get; set; }
    }
}

File: UI/travle_core/lib/src/models/favorite_toggle_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'favorite_toggle_response.g.dart';

/// The result of a favorite toggle (mirrors the backend `FavoriteToggleResponse`).
/// [isFavorite] is the state after the toggle — the client flips the heart from
/// this rather than assuming the outcome.
@JsonSerializable()
class FavoriteToggleResponse {
  FavoriteToggleResponse({
    required this.targetType,
    required this.targetId,
    required this.isFavorite,
  });

  /// "Destination" or "Tour".
  final String targetType;
  final int targetId;

  /// True if the target is now favorited, false if it was just removed.
  final bool isFavorite;

  factory FavoriteToggleResponse.fromJson(Map<String, dynamic> json) =>
      _$FavoriteToggleResponseFromJson(json);

  Map<String, dynamic> toJson() => _$FavoriteToggleResponseToJson(this);
}
